cognitive/src/main/scala/com/microsoft/ml/spark/cognitive/TextTranslator.scala

```diff
@@ -387,20 +488,66 @@ trait HasTextAndTranslationInput extends HasServiceParams {
 object DictionaryExamples extends ComplexParamsReadable[DictionaryExamples]
 
 class DictionaryExamples(override val uid: String) extends TextTranslatorBase(uid)
-  with HasTextAndTranslationInput with HasFromLanguage with HasToLanguage with BasicLogging {
+  with HasTextAndTranslationInput with HasFromLanguage with HasToLanguage
+  with HasCognitiveServiceInput with BasicLogging {
   logClass()
 
   def this() = this(Identifiable.randomUID("DictionaryExamples"))
 
   def urlPath: String = "dictionary/examples"
 
-  override protected def prepareEntity: Row => Option[AbstractHttpEntity] = {
-    r =>
-      Some(new StringEntity(
-        getValue(r, textAndTranslation).asInstanceOf[Seq[Row]]
-          .map(x => Map("Text" -> x.getString(0), "Translation" -> x.getString(1)))
-          .toJson.compactPrint, ContentType.APPLICATION_JSON))
+  override protected def inputFunc(schema: StructType): Row => Option[HttpRequestBase] = {
+    { row: Row =>
+      if (shouldSkip(row)) {
+        None
+      } else {
+        val urlParams: Array[ServiceParam[Any]] =
+          getUrlParams.asInstanceOf[Array[ServiceParam[Any]]]
+
+        val textAndTranslations = getValue(row, textAndTranslation)
+        if (textAndTranslations.isEmpty)
+          None
+        else {
+
+          val base = getUrl + "?api-version=3.0"
+          val appended = if (!urlParams.isEmpty) {
+            "&" + URLEncodingUtils.format(urlParams.flatMap(p =>
+              getValueOpt(row, p).map {
+                val pName = p.name match {
+                  case "fromLanguage" => "from"
+                  case "toLanguage" => "to"
+                  case s => s
+                }
+                v => pName -> p.toValueString(v)
+              }
+            ).toMap)
+          } else {
+            ""
+          }
+
+          val post = new HttpPost(base + appended)
+          getValueOpt(row, subscriptionKey).foreach(post.setHeader("Ocp-Apim-Subscription-Key", _))
+          getValueOpt(row, subscriptionRegion).foreach(post.setHeader("Ocp-Apim-Subscription-Region", _))
+          post.setHeader("Content-Type", "application/json; charset=UTF-8")
+
+          val json = textAndTranslations.head.getClass.getTypeName match {
+            case "scala.Tuple2" => textAndTranslations.map(
```

Comment:
This looks a little fishy, is there a reason we cant use regular pattern matching here?

Reply:
Yes, because we added `def setTextAndTranslation(v: (String, String)): this.type = setScalarParam(textAndTranslation, Seq(v))` function and in this case even though in `getValue` function we cast it into Seq[(String, String)] the underlying type is still Tuple2, and for the Tuple case I can't cast it into Seq[Row] and use `Map("Text" -> s.getString(0), "Translation" -> s.getString(1))` directly...